src/main/java/com/devonfw/quarkus/productmanagement/ApplicationEntity.java

```diff
@@ -0,0 +1,8 @@
+package com.devonfw.quarkus.productmanagement;
+
+/**
+ * This is the abstract interface for a {@link GenericEntity}. We are using {@link Long} for all {@link #getId() primary
+ * keys}.
+ */
+public abstract interface ApplicationEntity extends GenericEntity<Long> {
```

Comment:
I think you don't need `ApplicationEntity` and `GenericEntity`. Just let your entity extend from ApplicationPersistenceEntity